vars/piperStageWrapper.groovy

```diff
@@ -0,0 +1,68 @@
+import com.sap.piper.Utils
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.ConfigurationLoader
+import groovy.transform.Field
+
+@Field String STEP_NAME = 'piperStageWrapper'
+
+void call(Map parameters = [:], body) {
+
+    def script = parameters.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+    def utils = parameters.juStabUtils ?: new Utils()
+
+    def stageName = parameters.stageName?:env.STAGE_NAME
+
+    // load default & individual configuration
+    Map config = ConfigurationHelper
+        .loadStepDefaults(this)
+        .mixin(ConfigurationLoader.defaultStageConfiguration(this, stageName))
+        .mixinGeneralConfig(script.commonPipelineEnvironment)
+        .mixinStageConfig(script.commonPipelineEnvironment, stageName)
+        .mixin(parameters)
+        .use()
+
+    stageLocking(config) {
+        withNode(config) {
+            try {
+                utils.unstashAll(config.stashContent)
+
+                def scriptFileName = stageExitFilePath(stageName, config)
+                if (fileExists(scriptFileName)) {
+                    Script interceptor = load(scriptFileName)
+                    echo "[${STEP_NAME}] Running interceptor '${scriptFileName}' for ${stageName}."
+                    //passing config twice to keep compatibility with https://github.com/SAP/cloud-s4-sdk-pipeline-lib/blob/master/vars/runAsStage.groovy
+                    interceptor(body, stageName, config, config)
```

Comment:
What about passing general and stage configuration separately? Merging can still be done in the stage.

Reply:
Do we have a need to distinguish between stage and general configuration in the exit?

With https://github.com/SAP/jenkins-library/pull/341/files#diff-107e29747c644fec9fcc5b5dac73bff6R16 we have the complete config object already available including all defaults.

Reply:
switched to passing `config` only once